tiled/client/stream.py

```diff
@@ -609,16 +641,38 @@ def data(self):
                 ).read()
         # Decode payload (bytes) into array.
         numpy_dtype = self.data_type.to_numpy_dtype()
-        return numpy.frombuffer(content, dtype=numpy_dtype).reshape(self.patch.shape)
+        if self.patch:
+            shape = self.patch.shape
+        else:
+            shape = self.shape
+        return numpy.frombuffer(content, dtype=numpy_dtype).reshape(shape)
+
+
+class LiveTableData(TableData):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    subscription: TableSubscription
+
+    def data(self):
+        "Get table"
+        # Registration occurs on import. Ensure this is imported.
+        from ..serialization import array
+
+        del array
```

Comment:
In a future version, I wonder if this pattern should be refactored into an explicit `register()` function that self-documents the intent.